src/blocked.c

```diff
@@ -291,6 +291,13 @@ void disconnectAllBlockedClients(void) {
  * be used only for a single type, like virtually any Redis application will
  * do, the function is already fair. */
 void handleClientsBlockedOnKeys(void) {
+
+    /* In case we are already in the process of unblocking clients we should
+     * not make a recursive call, in order to prevent breaking fairness. */
+    if (server.in_handling_blocked_clients)
```

Comment:
haha. I originally did that and thought it would be better to keep aligned with the existing way we handle such cases.
Can you say then why do we usually keep things in server? (I thought it was some kind of way to potentially have multiple servers...)

Reply:
i have no clue...
there are also a ton of places that use plain global variables in the top of C files (e.g. acl.c).
but in this case it can also be a static var in a function.
i guess the question is if we think it'll be useful for others or not.

Reply:
done